libvcx/src/aries/handlers/proof_presentation/prover/state_machine.rs

```diff
@@ -142,21 +141,16 @@ impl ProverSM {
             ProverState::PresentationPrepared(state) => {
                 match message {
                     ProverMessages::SendPresentation => {
-                        let connection_handle = connection_handle
-                            .ok_or(VcxError::from_msg(VcxErrorKind::ActionNotSupported, "Presentation is already sent"))?;
-                        connection::send_message(connection_handle, state.presentation.to_a2a_message())?;
+                        let send_message = send_message.unwrap();
```

Comment:
There is no need to unwrap here, we can return appropriate error from the `step` function (and l. 164) as well.